contracts/queue/src/contract.rs

```diff
@@ -105,7 +106,8 @@ fn dequeue<S: Storage, A: Api, Q: Querier>(
     let first = deps.storage.range(None, None, Order::Ascending)?.next();
 
     let mut res = HandleResponse::default();
-    if let Some((k, v)) = first {
+    if let Some(item) = first {
+        let (k, v) = item?;
```

Comment:
This usage is cleaner than that of queue, maybe we can use this above (combine last_key and my_key matches into one).

(These contracts are also supposed to be references on how to write code, which is why I care).

Reply:
Like this? cbaedce2cf2044277833741f22489806cae9d4e8